File: security/src/main/java/top/bettercode/summer/security/repository/ApiTokenRepository.java
package top.bettercode.summer.security.repository;

import org.springframework.lang.Nullable;
import org.springframework.security.core.userdetails.UserDetails;
import top.bettercode.summer.security.ApiToken;
import top.bettercode.summer.security.config.ApiSecurityProperties;

public interface ApiTokenRepository {

  void save(ApiToken authorization);

  void remove(ApiToken authorization);

  void remove(String scope, String username);

  default void validateUserDetails(UserDetails userDetails) {
  }

  default boolean needKickedOut(ApiSecurityProperties securityProperties, String scope,
      UserDetails userDetails) {
    return securityProperties.needKickedOut(scope);
  }


  @Nullable
  ApiToken findByScopeAndUsername(String scope, String username);

  @Nullable
  ApiToken findByAccessToken(String accessToken);

  @Nullable
  ApiToken findByRefreshToken(String refreshToken);

}

File: security/src/main/kotlin/top/bettercode/summer/security/repository/ApiTokenRepository.kt
package top.bettercode.summer.security.repository

import top.bettercode.summer.security.token.ApiToken

interface ApiTokenRepository {
    fun save(apiToken: ApiToken)
    fun remove(apiToken: ApiToken)
    fun remove(scope: String, username: String)
    fun remove(scope: String, usernames: List<String>)

    fun findByScopeAndUsername(scope: String, username: String): ApiToken?

    fun findByAccessToken(accessToken: String): ApiToken?

    fun findByRefreshToken(refreshToken: String): ApiToken?
}
